pkg/cloud/cloud_test.go

```diff
@@ -377,10 +378,21 @@ func checkResourceTolerations(t *testing.T, podSpec corev1.PodSpec) {
 		Operator: corev1.TolerationOpExists,
 		Effect:   corev1.TaintEffectNoSchedule,
 	}
+	noScheduleTaint := corev1.Toleration{
+		Operator: corev1.TolerationOpExists,
+		Effect:   corev1.TaintEffectNoSchedule,
+	}
 
 	tolerations := podSpec.Tolerations
-	assert.Contains(t, tolerations, uninitializedTaint, "PodSpec should tolerate the uninitialized taint")
-	assert.Contains(t, tolerations, notReadyTaint, "PodSpec should tolerate the not-ready taint")
+
+	g := NewWithT(t)
+	g.Expect(tolerations).To(SatisfyAny(
+		SatisfyAll(
```

Comment:
i'm just curious about the logic here, why the nested `SatisfyAll` inside the `SatisfyAny`? (i would have expected a `SatisfyAll` with all the tolerations)

Reply:
We previously required to have the Uninitialized and Unready taints tolerated, both of them together. The logic here now is saying that you must have `(Uninitiailized && Unready) || All`, or at least, that's what I've tried to do. Note the NoSchedule toleration is outside of the `SatisfyAll`

Reply:
ok, that  makes sense to me. it's slightly confusing to read, but i get what you are saying, thanks!